torchgeo/datasets/idtrees.py

```diff
@@ -0,0 +1,532 @@
+                labels = None
+            else:
+                geoms = self._load_geometries(directory)
+                labels = None
+
+        images = glob.glob(os.path.join(directory, "RemoteSensing", "RGB", "*.tif"))
+
+        return images, geoms, labels
+
+    def _load_labels(self, directory: str) -> Any:
+        """Load the csv files containing the labels.
+
+        Args:
+            directory: directory containing csv files
+
+        Returns:
+            a pandas DataFrame containing the labels for each image
+        """
+        import pandas as pd
+
+        path_mapping = os.path.join(directory, "Field", "itc_rsFile.csv")
+        path_labels = os.path.join(directory, "Field", "train_data.csv")
+        df_mapping = pd.read_csv(path_mapping)
+        df_labels = pd.read_csv(path_labels)
+        df_mapping = df_mapping.set_index("indvdID", drop=True)
+        df_labels = df_labels.set_index("indvdID", drop=True)
+        df = df_labels.join(df_mapping, on="indvdID")
+        df = df.drop_duplicates()
+        df.reset_index()
+        return df
+
+    def _load_geometries(self, directory: str) -> Dict[int, Dict[str, Any]]:
+        """Load the shape files containing the geometries.
+
+        Args:
+            directory: directory containing .shp files
+
+        Returns:
+            a dict containing the geometries for each object
+        """
+        filepaths = glob.glob(os.path.join(directory, "ITC", "*.shp"))
+
+        features: Dict[int, Dict[str, Any]] = {}
+        for path in filepaths:
+            with fiona.open(path) as src:
+                for i, feature in enumerate(src):
+                    if self.split == "train":
+                        features[feature["properties"]["id"]] = feature
+                    # Test set task 2 has no id
+                    else:
+                        features[i] = feature
+        return features
+
+    def _verify(self) -> None:
+        """Verify the integrity of the dataset.
+
+        Raises:
+            RuntimeError: if ``download=False`` but dataset is missing or checksum fails
+        """
+        url = self.metadata[self.split]["url"]
+        md5 = self.metadata[self.split]["md5"]
+        filename = self.metadata[self.split]["filename"]
+        directories = self.directories[self.split]
+
+        # Check if the files already exist
+        exists = [
+            os.path.exists(os.path.join(self.root, directory))
+            for directory in directories
+        ]
+        if all(exists):
+            return
+
+        # Check if zip file already exists (if so then extract)
+        filepath = os.path.join(self.root, filename)
+        if os.path.exists(filepath):
+            extract_archive(filepath)
+            return
+
+        # Check if the user requested to download the dataset
+        if not self.download:
+            raise RuntimeError(
+                "Dataset not found in `root` directory and `download=False`, "
```

Comment:
```suggestion
                f"Dataset not found in `root={self.root}` directory and `download=False`, "
```
This makes the error msg much more helpful if users accidentally specified the wrong directory

Reply:
True, however we aren't doing this in other datasets so we should probably make this change in a separate PR because it also requires updates to tests.

Reply:
We're doing it in CDL, which I think is where all this code was copied from. At some point we/me need to update the download methods for all packages.

Reply:
Lets do this in another PR